packages/redux-query/src/middleware/query.js

```diff
@@ -33,51 +60,99 @@ const defaultConfig = {
     statusCodes.SERVICE_UNAVAILABLE,
     statusCodes.GATEWAY_TIMEOUT,
   ],
-  getQueryKey,
 };
 
-const getPendingQueries = queries => {
-  return pickBy(queries, query => query.isPending);
+const getPendingQueries = (queries: QueriesState): QueriesState => {
+  const pendingQueries = {};
+
+  for (const queryKey in queries) {
+    if (queries.hasOwnProperty(queryKey)) {
+      const query = queries[queryKey];
+
+      if (query.isPending) {
+        pendingQueries[queryKey] = query;
+      }
+    }
+  }
+
+  return pendingQueries;
+};
+
+const pick = (source: { [key: string]: any }, keys: Array<string>): { [key: string]: any } => {
+  const picked = {};
+
+  for (const key of keys) {
+    if (source.hasOwnProperty(key)) {
+      picked[key] = source[key];
+    }
+  }
+
+  return picked;
 };
 
-const isStatusOK = status => status >= 200 && status < 300;
+const isStatusOk = (status: ?Status): boolean =>
+  status !== null && status !== undefined && status >= 200 && status < 300;
+
+const defaultTransform: Transform = (body: ?ResponseBody) => body || {};
 
-const queryMiddlewareAdvanced = networkInterface => (
-  queriesSelector,
-  entitiesSelector,
-  customConfig,
+const queryMiddleware = (
+  networkInterface: NetworkInterface,
+  queriesSelector: QueriesSelector,
+  entitiesSelector: EntitiesSelector,
+  customConfig: ?Config,
 ) => {
-  return ({ dispatch, getState }) => next => action => {
+  const networkHandlersByQueryKey: { [key: QueryKey]: NetworkHandler } = {};
+
+  const abortQuery = queryKey => {
+    const networkHandler = networkHandlersByQueryKey[queryKey];
+
+    if (networkHandler) {
+      networkHandler.abort();
+      delete networkHandlersByQueryKey[queryKey];
+    }
+  };
+
+  return ({ dispatch, getState }: ReduxStore) => (next: Next) => (action: PublicAction) => {
     let returnValue;
-    const { getQueryKey, ...config } = { ...defaultConfig, ...customConfig };
+    const config = { ...defaultConfig, ...customConfig };
+
     switch (action.type) {
```

Comment:
This code would be much more readable if every `switch` branch has been extracted to a separate function.

Reply:
Agreed! I think it's ok for now but after this is merged and you'd like to take a stab at refactoring it, I think that'd be a good change.

Reply:
Sure thing - I'd like :)